File: app/src/main/java/it/unical/mat/embasp/specializations/dlv2/desktop/DLV2DesktopService.java
package it.unical.mat.embasp.specializations.dlv2.desktop;

import java.util.List;

import it.unical.mat.embasp.base.Callback;
import it.unical.mat.embasp.base.InputProgram;
import it.unical.mat.embasp.base.OptionDescriptor;
import it.unical.mat.embasp.base.Output;
import it.unical.mat.embasp.platforms.desktop.DesktopService;
import it.unical.mat.embasp.specializations.dlv2.DLV2AnswerSets;

/** is an extention of {@link DesktopService} for DLV2 's solver */

public class DLV2DesktopService extends DesktopService {

	private final OptionDescriptor competitionOutputOption;

	public DLV2DesktopService(final String exe_path) {
		super(exe_path);
		load_from_STDIN_option = "--stdin";
		competitionOutputOption = new OptionDescriptor("--competition-output");
	}

	@Override
	protected Output getOutput(final String output, final String error) {
		return new DLV2AnswerSets(output, error);
	}

	@Override
	public void startAsync(final Callback callback, final List<InputProgram> programs, final List<OptionDescriptor> options) {
		super.startAsync(callback, programs, options);
	}

	@Override
	public Output startSync(final List<InputProgram> programs, final List<OptionDescriptor> options) {
		options.add(competitionOutputOption);
		return super.startSync(programs, options);
	}
}

File: CSharp version/it/unical/mat/embasp/specializations/dlv2/desktop/DLV2DesktopService.cs
using System.Collections.Generic;

namespace it.unical.mat.embasp.specializations.dlv2.desktop
{
    using Output = it.unical.mat.embasp.@base.Output;
    using DesktopService = it.unical.mat.embasp.platforms.desktop.DesktopService;

    public class DLV2DesktopService : DesktopService
    {
        public DLV2DesktopService(string exe_path) : base(exe_path)
        {
            load_from_STDIN_option = "--stdin";
        }

        protected internal override Output GetOutput(string output, string error) => new DLV2AnswerSets(output, error);
    }
}